operators/multiclusterobservability/controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -350,20 +372,44 @@ func (r *MultiClusterObservabilityReconciler) SetupWithManager(mgr ctrl.Manager)
 
 	cmPred := predicate.Funcs{
 		CreateFunc: func(e event.CreateEvent) bool {
-			if e.Object.GetName() == config.AlertRuleCustomConfigMapName &&
-				e.Object.GetNamespace() == config.GetDefaultNamespace() {
-				config.SetCustomRuleConfigMap(true)
-				return true
+			if e.Object.GetNamespace() == config.GetDefaultNamespace() {
+				if e.Object.GetName() == config.AlertRuleCustomConfigMapName {
+					config.SetCustomRuleConfigMap(true)
+					return true
+				} else if _, ok := e.Object.GetLabels()[config.BackupLabelName]; ok {
+					// resource already has backup label
+					return false
+				} else if _, ok := config.BackupResourceMap[e.Object.GetName()]; ok {
+					// resource's backup label must be checked
+					return true
+				} else if _, ok := e.Object.GetLabels()[config.GrafanaCustomDashboardLabel]; ok {
+					// ConfigMap with custom-grafana-dashboard labels, check for backup label
+					config.BackupResourceMap[e.Object.GetName()] = config.ResourceTypeConfigMap
+					return true
+				}
 			}
 			return false
 		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			// Find a way to restart the alertmanager to take the update
-			// if e.ObjectNew.GetName() == config.AlertRuleCustomConfigMapName &&
-			// 	e.ObjectNew.GetNamespace() == config.GetDefaultNamespace() {
-			// 	config.SetCustomRuleConfigMap(true)
-			// 	return e.ObjectOld.GetResourceVersion() != e.ObjectNew.GetResourceVersion()
-			// }
+			if e.ObjectNew.GetNamespace() == config.GetDefaultNamespace() {
+				if e.ObjectNew.GetName() == config.AlertRuleCustomConfigMapName {
+					// Grafana dynamically loads AlertRule configmap, nothing more to do
+					//config.SetCustomRuleConfigMap(true)
+					//return e.ObjectOld.GetResourceVersion() != e.ObjectNew.GetResourceVersion()
+					return false
+				} else if _, ok := e.ObjectNew.GetLabels()[config.BackupLabelName]; ok {
+					// resource already has backup label
+					return false
```

Comment:
same here. We should assume the scenario that user update the resource with backup label

Reply:
Per conversation earlier, the placement controller will still process them, so this code works as expected.